staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

```diff
@@ -1316,3 +1316,41 @@ type PartialObjectMetadataList struct {
 	// items contains each of the included items.
 	Items []PartialObjectMetadata `json:"items" protobuf:"bytes,2,rep,name=items"`
 }
+
+// Condition is a struct intended for direct embedding like
+// type FooStatus struct{
+//     // Represents the observations of a foo's current state.
+//     // +patchMergeKey=type
+//     // +patchStrategy=merge
+//     Conditions []DeploymentCondition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,1,rep,name=conditions"`
```

Comment:
Might also give an example of describing well-known types in the `Conditions []metav1.Condition` field doc, since consumers can't change field doc on the individual fields of the Condition struct directly.